cmd/admin-handlers-users.go

```diff
@@ -530,6 +530,9 @@ func (a adminAPIHandlers) AddServiceAccount(w http.ResponseWriter, r *http.Reque
 					errors.New("service accounts cannot be generated for temporary credentials without parent")), r.URL)
 				return
 			}
+			if targetUser == "" {
```

Comment:
Can't regular users have service accounts ? You might want to move this outside this block

Reply:
@vadmeste this is mainly for STS accounts that have parent user.